test/src/unit-cppapi-group.cc

```diff
@@ -323,58 +294,76 @@ TEST_CASE_METHOD(
 
   // Close group
   group.close();
-  remove_temp_dir(temp_dir);
 }
 
 TEST_CASE_METHOD(
-    GroupCPPFx, "C++ API: Group, set name", "[cppapi][group][read]") {
-  // Create and open group in write mode
-  // TODO: refactor for each supported FS.
-  std::string temp_dir = fs_vec_[0]->temp_dir();
-  create_temp_dir(temp_dir);
-
-  const tiledb::sm::URI array1_uri(temp_dir + "array1");
-  const tiledb::sm::URI array2_uri(temp_dir + "array2");
-  const tiledb::sm::URI array3_uri(temp_dir + "array3");
-  create_array(array1_uri.to_string());
-  create_array(array2_uri.to_string());
-  create_array(array3_uri.to_string());
-
-  tiledb::sm::URI group1_uri(temp_dir + "group1");
-  tiledb::Group::create(ctx_, group1_uri.to_string());
-
-  tiledb::sm::URI group2_uri(temp_dir + "group2");
-  tiledb::Group::create(ctx_, group2_uri.to_string());
+    GroupCPPFx,
+    "C++ API: Group, set name",
+    "[cppapi][group][read][rest-fails][sc-57867]") {
+  std::string array1_uri = vfs_test_setup_.array_uri("array1");
+  std::string array2_uri = vfs_test_setup_.array_uri("array2");
+  std::string array3_uri = vfs_test_setup_.array_uri("array3");
+  std::cerr << array1_uri << std::endl;
+
+  create_array(array1_uri);
+  create_array(array2_uri);
+  create_array(array3_uri);
+
+  std::string group1_uri = vfs_test_setup_.array_uri("group1");
+  std::string group2_uri = vfs_test_setup_.array_uri("group2");
+  tiledb::Group::create(ctx_, group1_uri);
+  tiledb::Group::create(ctx_, group2_uri);
+  std::cerr << group1_uri << std::endl;
 
   // Set expected
+  auto uri_format = [&](std::string uri) {
+    return vfs_test_setup_.is_rest() ? vfs_test_setup_.with_prefix(uri) : uri;
```

Comment:
Just an idea from looking at CI.. maybe this is flipped around and we want to use `with_prefix` for the `false` case?

Reply:
I removed all that `with_prefix` logic that was confusing and went back to using URIs.